.github/workflows/snyk.yml

```diff
@@ -0,0 +1,50 @@
+# GitHub action to run Snyk check for vulnerabilities in all requirement files.
+
+name: run-snyk
+
+on:
+  pull_request:
+  push:
+    branches: [main]
+
+jobs:
+  snyk:
+    runs-on: ubuntu-latest
+    permissions:
+      pull-requests: write
+    strategy:
+      matrix:
+        os: [ubuntu-latest]
+        python-version: ["3.10"]
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        id: setup_python
+        with:
+          python-version: ${{ matrix.python-version}}
+
+      - name: Install dependencies
+        shell: bash
+        run: |
+          pip install invoke
+          pip install -r requirements.txt -r test-requirements.txt -r dev-requirements.txt -r gpu-requirements.txt
+
+      - name: Install snyk
+        run: npm install -g snyk
+
+      - name: Run Snyk to check for vulnerabilities
+        id: snyk
+        env:
+          SNYK_TOKEN: ${{ secrets.SNYK_TOKEN }}
+        continue-on-error: false
+        run: |
+          if inv snyk; then
+            echo "Snyk check passed"
+            echo "snyk_failed=0" >> $GITHUB_OUTPUT
+          else
+            echo "Snyk check failed"
+            echo "snyk_failed=1" >> $GITHUB_OUTPUT
+            exit 1
+          fi
```

Comment:
I assume this should be enough?
```suggestion
        run: |
          inv snyk
```

I would probably convert the command to inv check-packages (in case we switch out snyk).

Reply:
inv scan-for-vulnerabities

Reply:
i'm good with the name change